Initial Commit

garyanewsome opened the pull request:
Initial push, for desk checking and trouble shooting purposes,

garyanewsome committed:
Initial Commit

Initial push, for desk checking and trouble shooting purposes,

ninjapanzer commented on src/assignment/pkg9/DiceFileReader.java:
```diff
@@ -18,33 +18,39 @@
+    // Put you read logic and populate lines
+    while (this.input.hasNext()) {
+      String line = this.input.nextLine();
+      if (line.charAt(0) != '#') { // excludes commented header lines
```
:heart:

ninjapanzer commented on src/assignment/pkg9/DiceReaderTest.java:
```diff
@@ -5,18 +5,63 @@
-	System.out.println(diceReader.getLines().toString().equals("[]"));
+
+  public static void test_reader() {
+    DiceFileReader diceReader = new DiceFileReader("dice.txt");
```
I always have to do a `./` infront of my local files so `"./dice.txt"`

but if this works ignore me

ninjapanzer commented on src/assignment/pkg9/DiceReaderTest.java:
```diff
@@ -5,18 +5,63 @@
+      String[] fileLine = lines.get(i).split(" ");
+
+      // seperate the file's line into die
+      for (int ii = 0; ii < fileLine.length; ii++) {
```
I never thought of using `ii` I kind of like it. We always learned to use `i` then `j` then `k`. Just a comment no change needed.

garyanewsome replied:
See, I never knew why everyone always used `i`. I was always taught use whatever you want, never knew of the `i`,`j`, `k` thing.

ninjapanzer commented on src/assignment/pkg9/DiceReaderTest.java:
```diff
@@ -5,18 +5,63 @@
+
+      // seperate the file's line into die
+      for (int ii = 0; ii < fileLine.length; ii++) {
+        String[] dieConfig = fileLine[ii].split(":");
```
good :+1:

ninjapanzer commented on src/assignment/pkg9/DiceReaderTest.java:
```diff
@@ -5,18 +5,63 @@
+          loadedSide = Integer.parseInt(dieConfig[2]);
+        }
+
+        switch (dieType) { // to be facied up once working
```
This looks really good!

ninjapanzer commented on src/assignment/pkg9/DiceReaderTest.java:
```diff
@@ -5,18 +5,63 @@
+        String dieType = dieConfig[1];
+        
+        int loadedSide = 0;
+        if (dieType.equals("loaded"));
```
No `;` here and this will work correctly

ninjapanzer commented on src/assignment/pkg9/DiceReaderTest.java:
```diff
@@ -5,18 +5,63 @@
+      }
+    }
+    // System.out.println(diceReader.getLines().toString().equals("[]"));
+    System.out.println(trayValue);
```
I think you need to reset tray value lower in the loop structure

garyanewsome committed:
Corrections

Bloody typo's and ; 's will get you everytime.

garyanewsome committed:
IT WORKS!!!

Ta Da!

srtinkey commented on main/src/assignment/pkg9/DiceReaderTest.java:
```diff
@@ -0,0 +1,66 @@
+
+      // seperate the file's line into die
+      for (int ii = 0; ii < fileLine.length; ii++) {
+        String[] dieConfig = fileLine[ii].split(":");
```
I like the use of i and ii in your for loops. A third nested loop could use iii, I would assume. It kinda ties them together and tells you where in the loops you are. I think I'm going to steal your idea.  :-)

srtinkey commented on main/src/assignment/pkg9/DiceReaderTest.java:
```diff
@@ -0,0 +1,66 @@
+        int loadedSide = 0;
+        if (dieType.equals("loaded")){
+          loadedSide = Integer.parseInt(dieConfig[2]);
+        }
```
Couldn't this statement for loadedSide be put in the switch case "loaded" block?

sshurie commented on main/src/assignment/pkg9/DiceFileReader.java:
```diff
@@ -0,0 +1,56 @@
+    while (this.input.hasNext()) {
+      String line = this.input.nextLine();
+      if (line.charAt(0) != '#') { // excludes commented header lines
+        this.lines.add(line);
```
Clever! I removed the lines in my text document, I didn't think to search through it like this. I'll have to go back and fix it!

garyanewsome committed:
KABOOM!

Blew up the program, made a mess of things, yet strangely made it
better. Still working out the details. Still need to write to a file.
Commit for B/U purposes.

garyanewsome committed:
I believe we have success.

Program works and writes output to a file. I left it so the program will
also print to the IDE, easier for checking results.

I feel like it could have been broken down further even into more
objects or methods.

garyanewsome committed:
TryCatch_Fix

Rewrote Try Catch statement to enclose only the statement and not the
block

kavicm commented on main/src/main/main.java:
```diff
@@ -0,0 +1,104 @@
+    }
+
+  }
+}
```
I had so much trouble with this assignment.  Your setup made the most sense to me.  I tried to follow your flow, but I still could not get mine to run.
